third_party/txt/src/skia/paragraph_builder_skia.cc

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "paragraph_builder_skia.h"
+#include "paragraph_skia.h"
+
+#include "third_party/skia/modules/skparagraph/include/ParagraphStyle.h"
+#include "third_party/skia/modules/skparagraph/include/TextStyle.h"
+#include "txt/paragraph_style.h"
+
+namespace skt = skia::textlayout;
+
+namespace txt {
+
+namespace {
+
+SkFontStyle MakeSkFontStyle(txt::FontWeight font_weight,
+                            txt::FontStyle font_style) {
+  return SkFontStyle(static_cast<SkFontStyle::Weight>(
+                         static_cast<int>(font_weight) * 100 + 100),
+                     SkFontStyle::Width::kNormal_Width,
+                     font_style == txt::FontStyle::normal
+                         ? SkFontStyle::Slant::kUpright_Slant
+                         : SkFontStyle::Slant::kItalic_Slant);
+}
+
+skt::ParagraphStyle TxtToSkia(const ParagraphStyle& txt) {
+  skt::ParagraphStyle skia;
+  skt::TextStyle text_style;
+
+  text_style.setFontStyle(MakeSkFontStyle(txt.font_weight, txt.font_style));
+  text_style.setFontSize(SkDoubleToScalar(txt.font_size));
+  text_style.setHeight(SkDoubleToScalar(txt.height));
+  text_style.setFontFamilies({SkString(txt.font_family.c_str())});
+  text_style.setLocale(SkString(txt.locale.c_str()));
+  skia.setTextStyle(text_style);
+
+  // TODO: strut
```

Comment:
Shouldn't the TODO comment be a full sentence?

Reply:
removed this